web/modules/custom/rdf_entity_provenance/src/ProvenanceHelperInterface.php

```diff
@@ -0,0 +1,52 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\rdf_entity_provenance;
+
+use Drupal\rdf_entity\RdfInterface;
+
+/**
+ * Provides helper methods to fetch, check and update provenance data.
+ */
+interface ProvenanceHelperInterface {
+
+  /**
+   * Checks whether the the entity has provenance records.
+   *
+   * If an entity has provenance records it means that it is associated with an
+   * external source and thus has been federated.
+   *
+   * @param \Drupal\rdf_entity\RdfInterface $rdf_entity
+   *   The rdf entity.
+   *
+   * @return bool
+   *   Whether the entity has provenance record associated and thus, is
+   *   federated.
+   */
+  public function isFederated(RdfInterface $rdf_entity): bool;
+
+  /**
+   * Loads or creates a provenance record for the passed rdf entity.
+   *
+   * @param \Drupal\rdf_entity\RdfInterface $rdf_entity
+   *   The rdf entity.
+   *
+   * @return \Drupal\rdf_entity\RdfInterface|null
+   *   The provenance activity related to the rdf entity passed.
+   */
+  public function getProvenanceActivity(RdfInterface $rdf_entity): RdfInterface;
```

Comment:
What if I want to get the provenance for a blacklisted entity? How do I pass the `$rdf_entity`? I guess we have to pass the ID (uri) of the entity to cover also assets not imported on our system (AKA not having Drupal entities). Also the naming should reflect, I think, the fact that we are passing the referred entity, like `getProvenanceByReferredEntity()`?